### PR TITLE
ci: send test coverage data to SonarQube/SonarCloud (SDKCF-4730)

### PR DESCRIPTION
**Pull request by imaginaris**
# Description
* Used slather to generate sonar generic format test report and added it to SonarQube and SonarCloud properties
* Added Tests directory to exclusions (@apparition47's contribution) 

## Links
SDKCF-4730

# Checklist
- [X] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [X] I removed all sensitive data **before every commit**, including API endpoints and keys
- [X] I ran `fastlane ci` without errors

### CODE REVIEW

**Comment by imaginaris** [comment]
@donnie-jp Sonarcloud's Auto scan feature we use, doesn't support code coverage information 😞 
https://sonarcloud.io/documentation/advanced-setup/automatic-analysis/

**Comment by donnie-jp** [comment]
> @donnie-jp Sonarcloud's Auto scan feature we use, doesn't support code coverage information 😞 https://sonarcloud.io/documentation/advanced-setup/automatic-analysis/

Yep I was aware of that limitation. Shame.

File: fastlane/Fastfile [hunk 1]
```diff
@@ -45,6 +46,18 @@ platform :ios do
...
+      scheme: ENV['REM_FL_TESTS_SCHEME'] || 'Tests',
+      sonarqube_xml: true,
+      use_bundle_exec: true,
+      proj: ENV['REM_FL_TESTS_PROJECT'],
+      workspace: ENV['REM_FL_TESTS_WORKSPACE'],
+      binary_basename: ENV['REM_FL_TESTS_SLATHER_BASENAME'],
+      ignore: '*.{h,m}')
+  end
+
+  lane :run_xcov do
```
Comment (donnie-jp) [inline fastlane/Fastfile:60]: Does this still get used?

**Review by donnie-jp — APPROVED** [review]

File: fastlane/Fastfile (hunk 1 above)
Comment (imaginaris) [reply fastlane/Fastfile:60]: No - it's just for devs to be run locally if needed. I might as well delete this lane

**Pull request merged**